src/gfx/pal_spec.cpp

```diff
@@ -203,37 +205,49 @@ static T readLE(U const *bytes) {
 
 /*
  * **Appends** the first line read from `file` to the end of the provided `buffer`.
+ *
+ * @return true if a line was read.
  */
-static void readLine(std::filebuf &file, std::string &buffer) {
+[[gnu::warn_unused_result]] // Ignoring EOF is a bad idea.
+static bool
+    readLine(std::filebuf &file, std::string &buffer) {
 	// TODO: maybe this can be optimized to bulk reads?
 	for (;;) {
 		auto c = file.sbumpc();
 		if (c == std::filebuf::traits_type::eof()) {
-			return;
+			return false;
 		}
 		if (c == '\n') {
 			// Discard a trailing CRLF
 			if (!buffer.empty() && buffer.back() == '\r') {
 				buffer.pop_back();
 			}
-			return;
+			return true;
 		}
 
 		buffer.push_back(c);
 	}
 }
 
+#define requireLine(kind, file, buffer) \
+	do { \
+		if (!readLine(file, buffer)) { \
+			error(kind " palette file is shorter than expected"); \
```

Comment:
```suggestion
			error(kind " palette file is too short"); \
```

Reply:
I feel like “too short” is too vague: too short relative to what? Hence this choice of wording.